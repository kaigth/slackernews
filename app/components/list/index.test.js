import React from 'react';
import List from './index';
import { BrowserRouter as Router } from "react-router-dom";
import renderer from 'react-test-renderer';

it('renders the List correctly', () => {
  const testItems = [
    {
      by: 'SuperNeatGuy',
      descendants: 0,
      id: 1,
      score: 1,
      time: 1518982182,
      title: 'This is bat country!',
      type: 'story',
      url: 'http://skyward.io'
    },
    {
      by: 'MrFancyPants',
      descendants: 0,
      id: 2,
      score: 10,
      time: 1518982186,
      title: 'Wax on, wax off.',
      type: 'story',
      url: 'http://skyward.io'
    }
  ];

  const tree = renderer
    .create(<Router><List items={ testItems } /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});