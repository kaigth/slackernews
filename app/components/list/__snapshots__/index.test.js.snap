// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the List correctly 1`] = `
<div
  className="list"
>
  <div
    className="_item"
  >
    <p
      className="_title"
    >
      This is bat country!
    </p>
    <div
      className="_detail"
    >
      <div
        className="_score"
      >
        1 point
      </div>
      <div
        className="_user"
      >
        By: 
        SuperNeatGuy
      </div>
      <div
        className="_time"
      >
        When: 
        02/18/2018
      </div>
    </div>
    <a
      className="_link"
      href="http://skyward.io"
      onClick={[Function]}
      target="_blank"
    />
  </div>
  <div
    className="_item"
  >
    <p
      className="_title"
    >
      Wax on, wax off.
    </p>
    <div
      className="_detail"
    >
      <div
        className="_score"
      >
        10 points
      </div>
      <div
        className="_user"
      >
        By: 
        MrFancyPants
      </div>
      <div
        className="_time"
      >
        When: 
        02/18/2018
      </div>
    </div>
    <a
      className="_link"
      href="http://skyward.io"
      onClick={[Function]}
      target="_blank"
    />
  </div>
</div>
`;
